rtt_ros2_params/src/rtt_ros2_params.cpp

```diff
@@ -0,0 +1,296 @@
+    RTT::log(RTT::Debug) <<
+      getOwner()->provides()->getName() <<
+      " has a property " << property_name << RTT::endlog();
+  } else {
+    RTT::log(RTT::Info) <<
+      getOwner()->provides()->getName() <<
+      " has NO property " << property_name << RTT::endlog();
+    for (const auto property : getOwner()->properties()->getProperties() ) {
+      RTT::log(RTT::Debug) <<
+        "Available property: " << property->getName() <<
+        RTT::endlog();
+    }
+  }
+
+  auto prop = getOwner()->properties()->getProperty(property_name);
+  if (nullptr == prop) {
+    // Try to find it among orphan parameters (previously loaded, without a component interface)
+    if (orphan_params_.find(property_name) == orphan_params_.end()) {
+      // When totally new, then create an orphan property, i.e. owned by the ros2-params service
+      orphan_params_[property_name] = paramvalue;
+      this->addProperty(property_name, orphan_params_[property_name])
+      .doc("Property loaded from ROS2 param: " + param_name);
+      prop = getProperty(property_name);
+    } else {
+      orphan_params_[property_name] = paramvalue;
+      prop = getProperty(property_name);
+    }
+    return true;
+  } else {
+    RTT::Property<rclcpp::ParameterValue> prop_paramvalue = prop;
+
+    // The property exists on the Owner's interface
+    RTT::log(RTT::Debug) <<
+      "The property " << property_name << " existed" <<
+      RTT::endlog();
+    if (!prop_paramvalue.ready()) {
+      RTT::log(RTT::Debug) <<
+        "But " << property_name <<
+        " cannot be casted into rclcpp::ParamValue" << RTT::endlog();
+
+      const auto property_ds = prop->getDataSource();
+      RTT::internal::ReferenceDataSource<rclcpp::ParameterValue> param_rds(paramvalue);
+
+      try {
+        if (property_ds->update(&param_rds)) {
+          // conversion successful
+          RTT::log(RTT::Debug) <<
+            "Property " <<
+            getOwner()->provides()->getName() << "." << property_name <<
+            " loaded successfully from ROS2 parameter " << param_name << RTT::endlog();
+          return true;
+        } else {
+          // conversion failed
+          RTT::log(RTT::Warning) <<
+            "Property " <<
+            getOwner()->provides()->getName() << "." << property_name <<
+            " failed to convert from ROS2 parameter " << param_name << RTT::endlog();
+          return false;
+        }
+      } catch (std::exception e) {
+        RTT::log(RTT::Error) <<
+          "The property " << getOwner()->provides()->getName() << "." <<
+          property_name << " could not be converted from ROS2 parameter " << param_name <<
+          ". Exception caught: " << e.what() << RTT::endlog();
+        return false;
+      }
+      return false;
+    } else {
+      // This is the rare case in which the Property was already of type rclcpp::ParameterValue
+      prop_paramvalue.set(paramvalue);
+    }
```

Comment:
You do not have to treat this case as a special case. The `property_ds->update(&param_rds)` method works in both cases, even if the types match exactly.